typescript/alert-center/src/utilities.ts

```diff
@@ -0,0 +1,165 @@
+    let newAssignment = false;
+    if (!employees.length) {
+      await ctx.client<Employee>('employee').insert({employee_name, alert_id: null, expiration: null});
+      employees = await ctx.client<Employee>('employee').where({employee_name}).select();
+    }
+    let alert : AlertEmployee[] = [];
+    if (employees[0].alert_id && more_time) {
+      // Extend time
+      ctx.logger.info(`Extending time for ${employee_name} on ${employees[0].alert_id}`);
+      if (employees[0].expiration?.getTime() ?? 0 < expiration.getTime()) {
+        employees[0].expiration = expiration;
+        await ctx.client<Employee>('employee').where({employee_name}).update({expiration});
+      }
+    } 
+    else if (employees[0].alert_id) {
+      alert = await ctx.client<AlertEmployee>('alert_employee').where({alert_id: employees[0].alert_id}).select();
+      return {employee: employees[0], newAssignment, alert};
+    }
+    else {
+      // Try to find assignment
+      const op = await ctx.client<AlertEmployee>('alert_employee').whereNull('employee_name').orderBy(['alert_id']).first();
+      if (op) {
+        op.employee_name = employee_name;
+        const alert_id = op.alert_id;
+        employees[0].alert_id = op.alert_id;
+        employees[0].expiration = expiration;
+        await ctx.client<Employee>('employee').where({employee_name}).update({alert_id, expiration});
+        await ctx.client<AlertEmployee>('alert_employee').where({alert_id}).update({employee_name});
+        newAssignment = true;
+        ctx.logger.info(`New Assignment for ${employee_name}: ${alert_id}`);
+      }
+    }
+    if (employees[0].alert_id) {
+      alert = await ctx.client<AlertEmployee>('alert_employee').where({alert_id: employees[0].alert_id}).select();
+    }
+    return {employee: employees[0], newAssignment, alert};
+  }
+
+  @Transaction()
+  static async employeeCompleteAssignment(ctx: KnexTransactionContext, employee_name: string) {
+    const employees = await ctx.client<Employee>('employee').where({employee_name}).select();
+    if (!employees.length) {
+      throw new Error(`No employee ${employee_name}`);
+    }
+    if (!employees[0].alert_id) {
+      throw new Error(`Employee ${employee_name} completed an assignment that did not exist`);
+    }
+    await ctx.client<AlertEmployee>('alert_employee').where({alert_id: employees[0].alert_id}).update({alert_status: AlertStatus.RESOLVED});
+    await ctx.client<Employee>('employee').where({employee_name}).update({alert_id: null, expiration: null});
+  }
+
+  @Transaction()
+  static async employeeAbandonAssignment(ctx: KnexTransactionContext, employee_name: string) {
+    const employees = await ctx.client<Employee>('employee').where({employee_name}).select();
+    if (!employees.length) {
+      throw new Error(`No employee ${employee_name}`);
+    }
+    if (!employees[0].alert_id) {
+      return; // Nothing to abandon
+    }
```

Comment:
Same as above.

Reply:
Same response - removed the first check but things complain without the second check